.github/workflows/test.yml

```diff
@@ -45,12 +105,105 @@ jobs:
         with:
           path: ./action/
 
-      - name: Test With Upload + Bonus Files
+      - name: Test With Upload No Hash
+        id: test
+        uses: ./action/
+        with:
+          no_commit_hash: true
+            
+      - name: Upload Artifact
+        id: upload
+        uses: actions/upload-artifact@v4
+        with:
+          name:  ${{ steps.test.outputs.name }}
+          path: "template/*"
+
+      - name: Add Artifact to Job Summary
+        run: 'echo "### 📦 Artifact url: ${{ steps.upload.outputs.artifact-url}}" >> $GITHUB_STEP_SUMMARY'
+
+      - name: Update Check Run
+        uses: actions/github-script@v7
+        if: always()
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            github.rest.checks.update({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              check_run_id: process.env.runId,
+              conclusion: '${{ job.status }}',
+            })
+
+
+  Test_LemLib_Without_Uploading_an_Artifact:
+    name: "Testing LemLib Without Uploading an Artifact"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Create Check Status
+        id: create-check
+        uses: actions/github-script@v7
+        with:
+          script: |
+            github.rest.checks.create({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              name: 'Test LemLib Without Uploading an Artifact',
+              head_sha: context.sha,
+              status: 'in_progress',
+              output: {
+                title: 'Test LemLib Without Uploading an Artifact',
+                summary: 'This test should pass'
+              }
+            }).then(result => core.exportVariable('runId', result.data.id))
+
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          repository: LemLib/LemLib
+          ref: 6fc0838e0aa1de03d7864002888f5d874b691910
+
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          path: ./action/
+
+      - name: Test Without Upload
         uses: ./action/
 
-  test-ez:
+      - name: Update Check Run
+        uses: actions/github-script@v7
+        if: always()
```

Comment:
why is there an if here? :sob:

Reply:
Because I want it to run even if the above jobs fail so that the check is actually updated and not indefinitely left as `in_progress`